readthedocs/core/static-src/core/js/doc-embed/sponsorship.js

```diff
@@ -3,224 +3,53 @@
-    var display_type = constants.PROMO_TYPES.LEFTNAV;
-    var priority = constants.DEFAULT_PROMO_PRIORITY;
+function create_ad_placement() {
     var selector = null;
     var class_name;         // Used for theme specific CSS customizations
-    var offset;
+    var container;
 
     if (rtd.is_mkdocs_builder() && rtd.is_rtd_like_theme()) {
         selector = 'nav.wy-nav-side';
-        class_name = 'ethical-rtd ethical-dark-theme';
+        class_name = 'dark raised';
     } else if (rtd.is_rtd_like_theme()) {
         selector = 'nav.wy-nav-side > div.wy-side-scroll';
-        class_name = 'ethical-rtd ethical-dark-theme';
+        class_name = 'dark raised';
     } else if (rtd.is_alabaster_like_theme()) {
         selector = 'div.sphinxsidebar > div.sphinxsidebarwrapper';
-        class_name = 'ethical-alabaster';
-    }
-
-    if (selector) {
-        $('<div />').attr('id', element_id)
-            .addClass(class_name).appendTo(selector);
-
-        // Determine if this element is above the fold
-        offset = $('#' + element_id).offset();
-        if (!offset || offset.top > $(window).height()) {
-            // If this is off screen, lower the priority
-            priority = constants.LOW_PROMO_PRIORITY;
-        }
-
-        return {
-            'div_id': element_id,
-            'display_type': display_type,
-            'priority': priority,
-        };
-    }
-
-    return null;
-}
-
-/*
- *  Creates a sidebar div where an ad could go
- *  Returns the ID of the div or none if no footer ad is possible
- */
-function create_footer_placement() {
-    var element_id = 'rtd-' + (Math.random() + 1).toString(36).substring(4);
-    var display_type = constants.PROMO_TYPES.FOOTER;
-    var priority = constants.DEFAULT_PROMO_PRIORITY;
-    var selector = null;
-    var class_name;
-    var offset;
-
-    if (rtd.is_rtd_like_theme()) {
-        selector = $('<div />').insertAfter('footer hr');
-        class_name = 'ethical-rtd';
-    } else if (rtd.is_alabaster_like_theme()) {
-        selector = 'div.bodywrapper .body';
-        class_name = 'ethical-alabaster';
+        class_name = 'raised';
     }
 
     if (selector) {
-        $('<div />').attr('id', element_id)
-            .addClass(class_name).appendTo(selector);
-
-        // Determine if this element is above the fold
-        offset = $('#' + element_id).offset();
-        if (!offset || offset.top < $(window).height()) {
-            // If the screen is short, lower the priority
-            // We don't want the ad to take up too much of the screen
-            priority = constants.LOW_PROMO_PRIORITY;
-        }
-
-        return {
-            'div_id': element_id,
-            'display_type': display_type,
-            'priority': priority,
-        };
+        container = $('<div />').attr('style', 'text-align:center').appendTo(selector);
+        $('<div />')
+            .attr('data-ea-publisher', "readthedocs")
```

Comment:
We could include the publisher group in the footer response to adjust this for our revshare folks, but that does require blocking this on the server response.

Reply:
I don't think you can initialize ads without first getting the project/user data. All the decisions you want to make require that data.

Reply:
One idea: 99% of our pageviews aren't revshare or gold users, so I still think loading the ad JS makes sense at pageload, but we can hide the ad display until we get the footer data back. That will keep ads at the same latency as now, and we just need to re-request them on revshare projects. We could even dump some metadata into the built HTML for revshare projects to be able to read that data prior to the initial request. 

This seems like it'll get us almost every ad view in the same latency as the previous approach, if not faster.